File: examples/prog2.py
a = [2 * i for i in range(100_000)]
b = [2 * i + 1 for i in range(100_000)]

def func(a, b):
    tamanho_a = len(a)
    tamanho_b = len(b)
    posicao_a = 0
    posicao_b = 0
    valor_a = a[posicao_a]
    valor_b = b[posicao_b]
    while True:
        if valor_a == valor_b:
            return True
        if valor_a < valor_b:
            posicao_a += 1
            if posicao_a >= tamanho_a:
                return False
            valor_a = a[posicao_a]
        else:
            posicao_b += 1
            if posicao_b >= tamanho_b:
                return False
            valor_b = b[posicao_b]

func(a, b)

File: examples/prog2.rs
fn func(a: &Vec<usize>, b: &Vec<usize>) -> bool {
    let tamanho_a = a.len();
    let tamanho_b = b.len();
    let mut posicao_a = 0;
    let mut posicao_b = 0;
    let mut valor_a = a[posicao_a];
    let mut valor_b = b[posicao_b];
    loop {
        if valor_a == valor_b {
            return true;
        }
        if valor_a < valor_b {
            posicao_a += 1;
            if posicao_a >= tamanho_a {
                return false;
            }
            valor_a = a[posicao_a];
        } else {
            posicao_b += 1;
            if posicao_b >= tamanho_b {
                return false;
            }
            valor_b = b[posicao_b];
        }
    }
}

fn main() {
    let a: Vec<usize> = (0..1_000_000).map(|i| i * 2).collect();
    let b: Vec<usize> = (0..1_000_000).map(|i| i * 2 + 1).collect();

    func(&a, &b);
}
